enterprise/core-edge/src/main/java/org/neo4j/coreedge/raft/replication/tx/ReplicatedTransactionCommitProcess.java

```diff
@@ -111,15 +114,13 @@ transaction content does eventually get replicated (e.g. delayed on the network)
             }
             catch ( InterruptedException | TimeoutException  e )
             {
-                futureTxId.cancel( false );
+                log.info( "Retrying replication of: " + operationContext );
             }
             catch ( ReplicationFailedException | ExecutionException e )
```

Comment:
There is a case here where if ReplicationFailedException happens on a subsequent round, then we cannot give up.